### PR TITLE
AEA-1401 add `Dialogues` type in `aealite`

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Add `Dialogues` struct in `aealite`.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: libs/go/aealite/protocols/dialogues.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+package protocols
+
+import "aealite/helpers"
+
+type Dialogues struct {
+	selfAddress                Address
+	endStates                  helpers.Set
+	messageType                *ProtocolMessageInterface
+	dialougueType              *Dialogue
```
Comment (DavidMinarsch) [inline libs/go/aealite/protocols/dialogues.go:29]: Not sure this approach will make much sense in Go..let's see

File: libs/go/aealite/protocols/storage.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
+/* -*- coding: utf-8 -*-
```
Comment (marcofavorito) [inline libs/go/aealite/protocols/storage.go:1]: no persistent dialogue yet; I suppose we can leave it outside this PR

File: .github/workflows/workflow.yml [hunk 3]
```diff
@@ -85,13 +85,13 @@ jobs:
...
-#        with:
-#          version: v1.28
-#          working-directory: libs/go/aealite
+      - name: Golang code style check (aealite)
+        uses: golangci/golangci-lint-action@v1
+        env:
+          ACTIONS_ALLOW_UNSECURE_COMMANDS: true
+        with:
+          version: v1.28
+          working-directory: libs/go/aealite
```
Comment (marcofavorito) [inline .github/workflows/workflow.yml:94]: we now have golang linters check again

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**